src/units/helper.cpp

```diff
@@ -88,4 +89,70 @@ std::string unit_level_tooltip(const unit_type &type)
 	return unit_level_tooltip(type.level(), type.advances_to(), mod_advancements);
 }
 
+std::string maybe_inactive(const std::string& str, bool active)
+{
+	return (active ? str : markup::span_color(font::INACTIVE_COLOR, str));
+}
+
+std::string format_cost_string(int unit_recall_cost, const int team_recall_cost)
+{
+	std::stringstream str;
+
+	if(unit_recall_cost < 0) {
+		unit_recall_cost = team_recall_cost;
+	}
+
+	if(unit_recall_cost > team_recall_cost) {
+		str << markup::img("themes/gold.png") << markup::span_color("#ff0000", unit_recall_cost);
+	} else if(unit_recall_cost == team_recall_cost) {
+		str << markup::img("themes/gold.png") << unit_recall_cost;
+	} else if(unit_recall_cost < team_recall_cost) {
+		str << markup::img("themes/gold.png~GS()") << markup::span_color("#00ff00", unit_recall_cost);
+	}
+
+	return str.str();
+}
+
+std::string format_cost_string(int unit_cost)
+{
+	return formatter() << markup::img("themes/gold.png") << unit_cost;
+}
+
+std::string format_name_string(const std::string& str, const bool can_recruit)
+{
+	return can_recruit ? markup::span_color("#cdad00", str) : str;
```

Comment:
IMO this should be conditional for the caller (ie, markup if can recruit).

Reply:
you mean the conditonal check should be moved to the caller?

Reply:
and obviously delete this.

Reply:
removed, wasn't being used.